File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s) :
    stack = []

    for value in s :
        if not stack :
            stack.append(value)
        else :
            if stack[-1] == value :
                stack.pop()
            else :
                stack.append(value)

    if stack :
        return 0

    return 1

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;
class Solution
{
    public int solution(String s)
    {
        int answer = -1;
        Stack<String> stack = new Stack<>();
        for(int i=0; i<s.length(); i++){
            String now = s.substring(i,i+1);
            if(stack.size() != 0 && stack.peek().equals(now)){
                stack.pop();
            } else{
                stack.push(now);
            }
        if(stack.size() == 0){
            answer = 1;
        } else{
            answer = 0;
        }
            
        }

        return answer;
    }
}